config-service-impl/src/main/java/org/hypertrace/config/service/ConfigServiceGrpcImpl.java

```diff
@@ -69,17 +71,24 @@ public void upsertConfig(
   public void getConfig(
       GetConfigRequest request, StreamObserver<GetConfigResponse> responseObserver) {
     try {
-      ContextSpecificConfig config = configStore.getConfig(getConfigResourceContext(request));
-
+      ConfigResourceContext configResourceContext = getConfigResourceContext(request);
+      Optional<ContextSpecificConfig> maybeConfig = configStore.getConfig(configResourceContext);
+      // initialize empty config if not present to handle merges better
+      if (maybeConfig.isEmpty()) {
+        maybeConfig = Optional.of(emptyConfig(configResourceContext.getContext()));
```

Comment:
At this point, you know config is present. Don't separate presence checks and access like this, instead you could for example write:
```java

ContextSpecificConfig config = configStore.getConfig(configResourceContext)
  .orElse(emptyConfig(configResourceContext.getContext()));
```

Then all of your later accesses can be type safe.

Reply:
thanks i will keep this in mind while using optional